Sources/Sentry/SentryOptions.m

```diff
@@ -14,60 +14,86 @@ @implementation SentryOptions
     ];
 }
 
+- (instancetype)init
+{
+    if (self = [super init]) {
+        self.enabled = @NO;
+
+        // TODO: Remove this side effect
+        SentrySDK.logLevel = kSentryLogLevelError;
+        self.logLevel = kSentryLogLevelError;
+
+        self.debug = @NO;
+        self.maxBreadcrumbs = defaultMaxBreadcrumbs;
+        self.integrations = SentryOptions.defaultIntegrations;
+        self.sampleRate = @1;
+        self.enableAutoSessionTracking = @NO;
+        self.sessionTrackingIntervalMillis = [@30000 unsignedIntValue];
+        self.attachStacktrace = @NO;
+
+        // Set default release name
+        NSDictionary *infoDict = [[NSBundle mainBundle] infoDictionary];
+        if (nil != infoDict) {
+            self.releaseName =
+                [NSString stringWithFormat:@"%@@%@+%@", infoDict[@"CFBundleIdentifier"],
+                          infoDict[@"CFBundleShortVersionString"], infoDict[@"CFBundleVersion"]];
+        }
+    }
+    return self;
+}
+
 - (_Nullable instancetype)initWithDict:(NSDictionary<NSString *, id> *)options
                       didFailWithError:(NSError *_Nullable *_Nullable)error
 {
-    self = [super init];
-    if (self) {
+    if (self = [self init]) {
         [self validateOptions:options didFailWithError:error];
         if (nil != error && nil != *error) {
             [SentryLog
                 logWithMessage:[NSString stringWithFormat:@"Failed to initialize: %@", *error]
                       andLevel:kSentryLogLevelError];
             return nil;
         }
-
-        // If no user-defined release, use default.
-        if (nil == self.releaseName) {
```

Comment:
Question: Seems like we lost this check now?
Is it possible to define a `releaseName` using this overload now?

Do I understand correctly: We'll set all default values the override it with whatever was set here in the dic?